job/job.go

```diff
@@ -110,14 +113,45 @@ func (j *Job) UnmarshalJSON(b []byte) error {
 	j.AggrID = aggrID
 
 	cbURL, ok := tmp["callback_url"].(string)
-	if !ok {
-		return errors.New("callback_url must be a string")
+	if ok {
+		_, err = url.ParseRequestURI(cbURL)
+		if err != nil {
+			return errors.New("Could not parse callback URL: " + err.Error())
+		}
+		j.CallbackURL = cbURL
```

Comment:
The previous way this was written was more in line with Go's idioms. You always want to have the exceptional case (early return) at the top and be done with it.

Reply:
nvm